dockers/actions-plan-preview/planpreview.go

```diff
@@ -170,6 +184,15 @@ func makeCommentBody(event *githubEvent, r *PlanPreviewResult) string {
 		if app.ApplicationKind == "TERRAFORM" {
 			lang = "hcl"
 		}
+
+		l := utf8.RuneCountInString(app.PlanDetails)
+		if detailLen+int64(l) > reservedDetailMessagesLen {
+			fmt.Fprint(&b, detailsOmittedMessage)
+			detailLen += int64(utf8.RuneCountInString(detailsOmittedMessage))
+			continue
+		}
+
+		detailLen += int64(l)
```

Comment:
Other messages have been added to `b` before so we can use `b.String()` directly to calculate a more accurate value.

```suggestion
                var (
                         msg = app.PlanDetails
                         curLen = utf8.RuneCountInString(b.String())
                		detailsLen = utf8.RuneCountInString(msg)
		)
		if curLen + detailsLen > reservedDetailMessagesLen {
		        msg = detailsOmittedMessage
		}
		fmt.Fprintf(&b, detailsFormat, lang, msg)
```

Reply:
At first I was going to do that, but then I decided not to because I might not be able to include all the information in the app if I did.

My plan is to first divide the message into two parts: the detail part and the other part. I assumed that the detail part could be as long as I wanted, but the other parts would not be that long (assumed less than 2000 characters in this pr). Then, I set the limit of the github message to be the part to write the detail, excluding the part to write the other part(So we can write details at most `65536 - 2000` characters, and others at most `2000` characters). This way, no matter how the detail looks like, the other part can always be written (as long as the number of characters in the other part is not exceeded (in this pr this is `2000`)).

Reply:
Thanks for your explanation. I got your point. 👍